### ISSUE TITLE
Fix the headings hierarchy in the Styles sidebar

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#42496]
### Description

Follow-up to https://github.com/WordPress/gutenberg/issues/42373

The headings hierarchy in the Styles sidebar is less than ideal. It skips levels, doesn't follow a logical structure and it appears it was implemented more for visual purposes rather than for good semantics.

Suffice to say the first heading within the Styles sidebar is a H5, which doesn't make sense and it's totally disconnected from the overall headings hierarchy.

A good headings structure is not just about good semantics. Headings are also a navigational tool for screen reader users, as they can use dedicated commands to jump through headings, get a sense of the overall content, and navigate a page. This principle applies also to administration user interfaces, where headings can be used to identify the main parts of the interface.

### Step-by-step reproduction instructions

- Go to the Site editor.
- Open the Styles sidebar.
- Click 'Colors'.
- Check the 'Colors' text to the right of the back button is a H5 heading. This is incorrect.
- Check that 'Palette' and 'Elements' are H2 headings. This doesn't communicate a proper hierarchy.
- Click 'Buttons'.
- Check the 'Buttons' text to the right of the back button is a H5 heading. This is incorrect.
- Check the 'Text color' heading underneath 'Buttons' is a H4 heading. This is incorrect because 'Text color' should be a sub-heading of 'Buttons'.
- Check the all-caps text 'THEME' is a H2 heading. It should be a sub-heading of 'Text color'.
- Check the all-caps text 'DEFAULT' is a H2 heading. It should be a sub-heading of 'Text color'.
- Check 'Background color' is a H2 heading. It skips a heading level but it should probably be a H2 in the first place.
- Check the all-caps text 'THEME' is a H2 heading. It should be a sub-heading of 'Background color'.
- Check the all-caps text 'DEFAULT' is a H2 heading. It should be a sub-heading of 'Background color'.

Hint; Use the HeadingsMap browser extension to check the headings hierarchy. 

### Screenshots, screen recording, code snippet

Worth mentioning the HeadingsMap browser extension only highlights skipped heading levels, and in the screenshot below there are a few headings highlighted as errors. It doesn't evaluate whether a heading level is logically correct or not, which requires evaluation by a human.

<img width="737" alt="Screenshot 2022-07-18 at 12 07 53" src="https://user-images.githubusercontent.com/1682452/179494989-4f8e313a-6679-46ae-8ea6-a993b1bd3bef.png">

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
